### ISSUE TITLE
Casting not necessary

### ISSUE THREAD

**Issue by kyscg** [tensorflow/tensorflow#26612]
Tell me if I'm wrong, but in `tensorflow/lite/kernels/internal/reference/integer_ops/softmax.h`, is casting to <int 32> necessary? (Line 89 & 90)

**Comment by haozha111** [comment]
It should be necessary. C++ std::min and std::max expect to see two arguments with the same type. So we need to do an explicit casting to ensure this, otherwise there might be a compiler error (based on the compiler you use). For more context, please see:
https://blogs.msmvps.com/gdicanio/2017/05/22/subtle-bug-with-stdminmax-function-templates/

**Comment by kyscg** [comment]
Yes, that makes sense. Thanks
